host/sgx/calls.c

```diff
@@ -365,6 +365,31 @@ static oe_result_t _handle_call_host_function(
     return result;
 }
 
+static const char* oe_ocall_str(oe_func_t ocall)
+{
+    static const char* func_names[] = {"OE_OCALL_CALL_HOST_FUNCTION",
+                                       "OE_OCALL_GET_QE_TARGET_INFO",
+                                       "OE_OCALL_GET_QUOTE",
+                                       "OE_OCALL_GET_REVOCATION_INFO",
+                                       "OE_OCALL_GET_QE_ID_INFO",
+                                       "OE_OCALL_THREAD_WAKE",
+                                       "OE_OCALL_THREAD_WAIT",
+                                       "OE_OCALL_THREAD_WAKE_WAIT",
+                                       "OE_OCALL_MALLOC",
+                                       "OE_OCALL_REALLOC",
+                                       "OE_OCALL_FREE",
+                                       "OE_OCALL_WRITE",
+                                       "OE_OCALL_SLEEP",
+                                       "OE_OCALL_GET_TIME",
+                                       "OE_OCALL_BACKTRACE_SYMBOLS"
+                                       "OE_OCALL_LOG"};
+
+    if (ocall >= OE_OCALL_BASE && ocall < __OE_FUNC_MAX)
```

Comment:
As written, the compiler does not know the relationship between OE_COUNTOF(func_names) and __OE_FUNC_MAX. We could add the following assertion:
OE_STATIC_ASSERT(OE_OCALL_BASE + OE_COUNTOF(func_names) == __OE_FUNC_MAX);

Reply:
That's a nice way to catch we add some OCALL in the enum but not in func_names.

Reply:
Sounds good, will add!

Reply:
So that won't work, because __OE_FUNC_MAX = OE_ENUM_MAX, which is 0xffffffff. I was a bit puzzled by that and thought this might be a mistake, but that seems to be done consistently across enums. I don't understand why, I would have thought having the max of each actual enum would be preferable.

Anyway, let me know if you want me to do something else instead, I cannot think of a good alternative without removing the assignment on the max value.

Reply:
It looks like @mikbras made all enums to be 4 bytes by forcing the MAX values to be 0xffffffff in [this commit](https://github.com/microsoft/openenclave/commit/480b29ad6c6a1a918d6c8808ecdd2738f1bdbba2). Please check with him what was the motivation and whether we could make an exception for `__OE_FUNC_MAX`.

Reply:
@jxyang The motivation was to ensure a consistent integer size for all enums rather than have the compiler pick the integer type.

@achamayou 
We could add another enum value OE_OCALL_MAX that contains the desired value and use that in the assertion instead.

Reply:
@anakrish Done